WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -2323,6 +2323,9 @@ private void showLoadingProgress(boolean showProgress) {
 
     private void setIsUpdating(boolean isUpdating, UpdateAction updateAction) {
         if (!isAdded() || mIsUpdating == isUpdating) {
+            if (!isUpdating) {
+                mRecyclerView.setRefreshing(false);
+            }
```

Comment:
⚠️ I don't think that addresses the root cause and has the potential of introducing a issues/crash, since it's now using the `mRecyclerView` when the Fragment might not be added to the Activity.

I did a quick investigation and it looks like the problem is that we have only 1 `mIsUpdating` field that is being used by 2 different progress indicators (`showLoadingProgress` and `mRecyclerView.setRefreshing`) but only one of them uses it, depending on the `updateAction` conditional statement below.

What is happening, in practice, is that for P2s with small number of posts we are doing several `setIsUpdating` calls with both `REQUEST_OLDER` and `REQUEST_NEWER` so there's a point where the `NEWER` sets the Recycler progress bar to visible, but then an `OLDER` call comes with `isUpdating=false`, which doesn't hide the Recycler progress bar so when the actual call with `NEWER` and `isUpdating=false` comes we don't do anything because of the logic on line 2325, since `mIsUpdating` is now `false`.

Here are some logs I added before line 2325 showing what happened for me when I set the filter on a P2 with just 1 post:
```
isUpdating=true,  mIsUpdating=false, updateAction=REQUEST_NEWER, isAdded = true        <--- caused mIsUpdating to be true and the Recycler progress to be visible

isUpdating=true,  mIsUpdating=true,  updateAction=REQUEST_NEWER, isAdded = true

isUpdating=true,  mIsUpdating=true,  updateAction=REQUEST_OLDER, isAdded = true        <--- was ignored, since mIsUpdating == isUpdating, so the bottom progress was never shown

isUpdating=true,  mIsUpdating=true,  updateAction=REQUEST_OLDER, isAdded = true

isUpdating=false, mIsUpdating=true,  updateAction=REQUEST_OLDER, isAdded = true        <--- caused mIsUpdating to be false, but since is REQUEST_OLDER, it tried hiding the bottom progress (but it was not visible) and didn't touch the Recycler progress (stayed visible)

isUpdating=false, mIsUpdating=false, updateAction=REQUEST_NEWER, isAdded = true        <--- was ignored, since mIsUpdating == isUpdating, so the Recycler progress was never hidden

isUpdating=true,  mIsUpdating=false, updateAction=REQUEST_OLDER, isAdded = true        <--- caused mIsUpdating to be true and the bottom progress to be visible

isUpdating=true,  mIsUpdating=true,  updateAction=REQUEST_OLDER, isAdded = true

isUpdating=false, mIsUpdating=true,  updateAction=REQUEST_OLDER, isAdded = true        <--- caused mIsUpdating to be false and hid the bottom progress
```

I see you're AFK today so I will try to fix it by avoiding using a single field for both progress indicators.

Reply:
I just pushed some commits in this PR to fix this issue and hopefully make the progress indicator states more reliable as well. @daniloercoli can you also confirm it works for you and review the code?

Reply:
The test appears successful and everything seems to be functioning correctly. However, I recommend adding a synchronize block when we use to the newly introduced object `mCurrentUpdateActions` to prevent any potential race conditions that could occur when updating/reading it on two different threads. Will add a more detailed comment to the commit.